datafusion/expr/src/expr_rewriter/mod.rs

```diff
@@ -279,6 +279,57 @@ where
     expr.alias_if_changed(original_name)
 }
 
+/// Handles ensuring the name of rewritten expressions is not changed.
```

Comment:
FYI @findepi here is an example of using Expr::Alias as described in https://github.com/apache/datafusion/issues/1468#issuecomment-2308237517